spec/index.bs

```diff
@@ -2517,6 +2518,82 @@ dictionary GPUImageBitmapCopyView {
+
+</div>
+
+<div algorithm class=validusage>
+
+<dfn abstract-op>copyBufferToTexture Valid Usage</dfn>
+
+Given a {{GPUCommandEncoder}} |encoder| and the arguments {{GPUBufferCopyView}} |source|, {{GPUTextureCopyView}} |destination|, {{GPUExtent3D}} |copySize|, the following validation rules apply:
+
+  - |encoder| must be a [=valid=] {{GPUCommandEncoder}}.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPURenderPassEncoder}} is active on |encoder|.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPUComputePassEncoder}} is active on |encoder|.
+  - |source|.{{GPUBufferCopyView/buffer}} must be a [=valid=] {{GPUBuffer}}.
+  - |destination|.{{GPUTextureCopyView/texture}} must be a [=valid=] {{GPUTexture}}.
+  - The {{GPUBuffer/[[usage]]}} of |source|.{{GPUBufferCopyView/buffer}} must contain {{GPUBufferUsage/COPY_SRC}}.
+  - The {{GPUTexture/[[textureUsage]]}} of |destination|.{{GPUTextureCopyView/texture}} must contain {{GPUTextureUsage/COPY_DST}}.
+  - The {{GPUTexture/[[sampleCount]]}} of |destination|.{{GPUTextureCopyView/texture}} must be 1.
+  - If |source|.{{GPUBufferCopyView/rowsPerImage}} is not 0, it must be greater than or equal to |copySize|.{{GPUExtent3DDict/height}}.
+  - |source|.{{GPUBufferCopyView/rowsPerImage}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} must be a multiple of [=texel block width=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |copySize|.{{GPUExtent3DDict/width}} must be a multiple of [=texel block width=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |copySize|.{{GPUExtent3DDict/height}} must be a multiple of [=texel block height=] of the {{GPUTexture/[[format]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |source|.{{GPUBufferCopyView/bytesPerRow}} must be a multiple of 256.
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is not 0, it must be greater than or equal to the number of |bytes in a complete row|.
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is 0, the number of |bytes in a complete row| must be a multiple of 256.
+  - |destination|.{{GPUTextureCopyView/mipLevel}} must be less than the {{GPUTexture/[[mipLevelCount]]}} of |destination|.{{GPUTextureCopyView/texture}}.
+  - |destination|.{{GPUTextureCopyView/arrayLayer}} must be less than the {{GPUTexture/[[textureSize]]}}.{{GPUExtent3DDict/depth}} of |destination|.{{GPUTextureCopyView/texture}}.
```

Comment:
This is only true for 2d. For 1d and 3d it must be 0.

Reply:
Done